Source/WebKit/WebProcess/Network/WebTransportSendStreamSink.cpp

```diff
@@ -64,10 +67,38 @@ void WebTransportSendStreamSink::write(WebCore::ScriptExecutionContext& context,
 
     WTF::switchOn(bufferSource.releaseReturnValue(), [&](auto&& arrayBufferOrView) {
         constexpr bool withFin { false };
-        context.enqueueTaskWhenSettled(session->streamSendBytes(m_identifier, arrayBufferOrView->span(), withFin), WebCore::TaskSource::Networking, [promise = WTFMove(promise)] (auto&&) mutable {
-            promise.resolve();
+        context.enqueueTaskWhenSettled(session->streamSendBytes(m_identifier, arrayBufferOrView->span(), withFin), WebCore::TaskSource::Networking, [promise = WTFMove(promise)] (auto&& exception) mutable {
+            if (!exception)
+                promise.settle(Exception { ExceptionCode::NetworkError });
+            else if (*exception)
+                promise.settle(WTFMove(**exception));
+            else
+                promise.resolve();
         });
     });
 }
 
+void WebTransportSendStreamSink::close()
+{
+    ASSERT(RunLoop::isMain());
+    if (m_isClosed)
+        return;
+    RefPtr session = m_session.get();
+    if (session)
+        session->streamSendBytes(m_identifier, std::span<const uint8_t>(), true);
```

Comment:
We can probably just use { } for the span.  We might want to use a 2-state enum for WithFin to make the "true" easier to tell what it means to a future reader.